staging/src/k8s.io/apiserver/pkg/registry/generic/rest/streamer.go

```diff
@@ -61,11 +64,28 @@ func (s *LocationStreamer) InputStream(ctx context.Context, apiVersion, acceptHe
 	if transport == nil {
 		transport = http.DefaultTransport
 	}
+	requestAddress := s.Location.Host
+	klog.Infof("Sending request to %q.", requestAddress)
+
+	if httpTransport, ok := transport.(*http.Transport); ok {
+		networkContext := server.NetworkContext{ConnectivityServiceName: "cluster"}
+		contextDialer, err := server.Lookup(networkContext)
+		if err != nil {
+			return nil, false, "", fmt.Errorf("failed to get connection for %s, got %v", s.Location.String(), err)
+		}
+		httpTransport.DialContext = contextDialer
+	} else {
+		klog.Warningf("Failed to set proxy on transport as transport is type %T.", transport)
```

Comment:
soft failures if we don't recognize the transport type or the httpTransport.DialContext is already set seem like they would be really hard to notice... should this fail more aggressively in those cases?